src/query/block/types.go

```diff
@@ -62,6 +78,19 @@ type SeriesIter interface {
 	Meta() Metadata
 }
 
+// UnconsolidatedSeriesIter iterates through a block horizontally
+type UnconsolidatedSeriesIter interface {
+	Iterator
+	// Current returns the current series for the block
+	Current() (UnconsolidatedSeries, error)
+	// SeriesCount returns the number of series
+	SeriesCount() int
+	// SeriesMeta returns the metadata for each series in the block
+	SeriesMeta() []SeriesMeta
+	// Meta returns the metadata for the block
+	Meta() Metadata
```

Comment:
Can roll these into a single interface and share that with `SeriesIter`

Reply:
good point